dependencies/clients/teku.go

```diff
@@ -0,0 +1,349 @@
+package clients
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+
+	"github.com/lukso-network/tools-lukso-cli/common/errors"
+	"github.com/lukso-network/tools-lukso-cli/common/system"
+	"github.com/lukso-network/tools-lukso-cli/common/utils"
+	"github.com/lukso-network/tools-lukso-cli/flags"
+	"github.com/lukso-network/tools-lukso-cli/pid"
+)
+
+const (
+	tekuDepsFolder = "teku" // folder in which both teku and JDK are stored
+	tekuFolder     = "teku" // folder in which teku is stored (in tekuDepsFolder)
+	jdkFolder      = "jdk"  // folder in which JDK is stored (in tekuDepsFolder)
+	jdkInstallURL  = "https://download.java.net/java/GA/jdk20.0.2/6e380f22cbe7469fa75fb448bd903d8e/9/GPL/openjdk-20.0.2_|OS|-|ARCH|_bin.tar.gz"
+)
+
+type TekuClient struct {
+	*clientBinary
+}
+
+func NewTekuClient() *TekuClient {
+	return &TekuClient{
+		&clientBinary{
+			name:           tekuDependencyName,
+			commandName:    "teku",
+			baseUrl:        "https://artifacts.consensys.net/public/teku/raw/names/teku.tar.gz/versions/|TAG|/teku-|TAG|.tar.gz",
```

Comment:
Please extract to `dependencies/clients/teku.go` consts if possible

Reply:
Done